src/app/stack/stack-details/stack-details.component.ts

```diff
@@ -227,7 +234,100 @@ export class StackDetailsComponent implements OnChanges {
         this.componentFilterBy = filterBy.filterBy;
     }
 
-    constructor(private stackAnalysisService: StackAnalysesService) { }
+    modalRef: BsModalRef;
+    token: string;
+    tokenerror: TokenErrorModel = {
+        status: false,
+        type: '',
+        length: 0
+    };
+    tokenErrorStatus: boolean = true;
+    tokenDetail: TokenDetailModel = {
+        id: '',
+        status: 'freetier'
+    }
+    tokenAlertsMessage: string;
+
+
+    observableToken: Observable<any>;
+
+    constructor(private stackAnalysisService: StackAnalysesService, private modalService: BsModalService) { }
+
+    async submitToken() {
+        if (!this.tokenErrorStatus) {
+            await this.stackAnalysisService.putToken(this.getBaseUrl(this.stack), '4c2c13be-2f0b-44a0-9e86-18abb3c60a6b', this.token, this.gatewayConfig)
+                .then(res => {
+                    console.log(res.status);
+                })
+                .catch(err => {
+                    console.log(err);
+                });
+
+            console.log('refresh in 5 sec');
+            setTimeout(() => {
+                this.init()
+            }, 5000);
+        }
+
+    }
+
+    async setTokenStatus() {
+
+        // 3620266d-ae88-42ce-bbb0-699f2dd67c49
+        await this.stackAnalysisService.getTokenStatus(this.getBaseUrl(this.stack), '4c2c13be-2f0b-44a0-9e86-18abb3c60a6b', this.gatewayConfig)
+            .then(res => {
+                this.tokenDetail.id = res.id;
+                this.tokenDetail.status = res.status.toLowerCase();
+            })
+            .catch(err => {
+                console.log(err);
+            });
+
+        let resultDetails = this.tokenDetail.status;
+
+        switch (resultDetails.toLowerCase()) {
+            case 'freetier':
+                this.tokenAlertsMessage = 'Unregistered'
+                break;
+            case 'registered':
+                this.tokenAlertsMessage = 'Registered'
+                break;
+            case 'expired':
+                this.tokenAlertsMessage = 'Token Expired'
+                break;
+            default:
+                break;
+        }
+    }
+
+    openModal(template: TemplateRef<any>) {
+        this.modalRef = this.modalService.show(template);
+    }
+
+    checkToken() {
+        this.tokenerror = {
+            status: false,
+            type: '',
+            length: 0
+        };
+        this.tokenerror.length = this.token.length;
+        if (this.token.length === 0 || this.token == undefined) {
+            this.tokenerror.status = true;
+            this.tokenerror.type = '';
+        } else if (this.token.length !== 36) {
+            this.tokenerror.status = true;
+            this.tokenerror.type = "Token length should be of 36 character";
+        } else if (!/^[a-zA-Z0-9-]*$/.test(this.token)) {
```

Comment:
UUID is a hex string. Could you follow https://stackoverflow.com/a/13653180/1942688 for the proper regex?

Reply:
updated it.